R/bainCommon.R

```diff
@@ -0,0 +1,1288 @@
+  citations <- c(
+    "Gu, X., Mulder, J., and Hoijtink, H. (2017). Approximate adjusted fractional Bayes factors: A general method for testing informative hypotheses. British Journal of Mathematical and Statistical Psychology, 71, 229-261. DOI: 10.1111/bmsp.12110",
+    "Hoijtink, H., Mulder, J., van Lissa, C., and Gu, X. (2018). A Tutorial on testing hypotheses using the Bayes factor. Psychological Methods, 24, 539-556. DOI: 10.1037/met0000201.",
+    "Hoijtink, H., Gu, X., and Mulder, J. (2018). Bayesian evaluation of informative hypotheses for multiple populations. British Journal of Mathematical and Statistical Psychology, 72, 219-243. DOI: 10.1111/bmsp.12145"
+  )
+  return(citations)
+}
+
+# Create a container for the results
+.bainGetContainer <- function(jaspResults, deps) {
+  if (is.null(jaspResults[["bainContainer"]])) {
+    jaspResults[["bainContainer"]] <- createJaspContainer()
+    jaspResults[["bainContainer"]]$dependOn(options = deps)
+    jaspResults[["bainContainer"]]$position = 1
+  }
+  invisible(jaspResults[["bainContainer"]])
+}
+
+# Read the data set
+.bainReadDataset <- function(options, type, dataset){
+  
+  numerics <- switch(type,
+                     "onesampleTTest" = options[["variables"]],
+                     "pairedTTest" = unique(unlist(options[["pairs"]])),
+                     "independentTTest" = unlist(options[["variables"]]),
+                     "anova" = options[["dependent"]],
+                     "ancova" = c(options[["dependent"]], unlist(options[["covariates"]])),
+                     "regression" = c(options[["dependent"]], unlist(options[["covariates"]])),
+                     "sem" = NULL)
+  numerics <- numerics[numerics != ""]
+  factors <- switch(type,
+                    "onesampleTTest" = NULL,
+                    "pairedTTest" = NULL,
+                    "independentTTest" = options[["groupingVariable"]],
+                    "anova" = options[["fixedFactors"]],
+                    "ancova" = options[["fixedFactors"]],
+                    "regression" = NULL,
+                    "sem" = NULL)
+  factors <- factors[factors != ""]
+  vars <- c(numerics, factors)
+  
+  if(type != "sem"){
+    if (is.null(dataset)) {
+      trydata	<- .readDataSetToEnd(columns.as.numeric = numerics, columns.as.factor = factors)
+      missing	<- names(which(apply(trydata, 2, function(x) { any(is.na(x))} )))
+      dataset	<- .readDataSetToEnd(columns.as.numeric = numerics, columns.as.factor = factors, exclude.na.listwise = vars)
+      if((type == "anova" || type == "ancova") && options[["fixedFactors"]] != ""){
+        if(any(grepl(pattern = " ", x = levels(dataset[, options[["fixedFactors"]] ])))) {
+          jaspBase:::.quitAnalysis(gettext("Bain does not accept factor levels that contain spaces. Please remove the spaces from your factor levels to continue."))
+        }
+      }
+    } else {
+      dataset 	<- .vdf(dataset, columns.as.numeric = numerics, columns.as.factor = factors)
```

Comment:
`.vdf()` could go too

Reply:
What is its replacement?

Reply:
There is none, all encoding happens automatically before and after the analysis

Reply:
So that can just be `dataset <- c(numerics, factors)`?

Reply:
Ah, right, it also coerces the columns to the correct type. Never mind this comment, it should stay!